lib/rucio/daemons/judge/repairer.py

```diff
@@ -90,7 +91,11 @@ def rule_repairer(once=False):
 
             if not rules and not once:
                 logging.debug('rule_repairer[%s/%s] did not get any work (paused_rules=%s)' % (heartbeat['assign_thread'], heartbeat['nr_threads'], str(len(paused_rules))))
-                graceful_stop.wait(60)
+                end_time = time.time()
```

Comment:
I think it's worth creating an utils function with this code snippet and call it in all places instead of copy-pasting this code block

Reply:
Okay, so maybe I can create this function in `rucio.common.utils`:
```python
def sleep_time_daemons(start_time, sleep_time, graceful_stop):
    end_time = time.time()
    time_diff = end_time - start_time
    if time_diff < sleep_time:
        logging.info('Sleeping for a while :  %s seconds', (sleep_time - time_diff))
        graceful_stop.wait(sleep_time - time_diff)
```
And call it in each daemon:
```suggestion
                sleep_time_daemons(start, sleep_time, graceful_stop)
```

Reply:
That's what I had in mind. Nothing very fancy, just to avoid copy-paste. Maybe a different naming ? maybe daemon_sleep() ? I don't know, but sleep_time_daemons is confusing because it makes me think that it will compute a time; not that it will sleep.